sound/soc/sof/intel/hda-dai.c

```diff
@@ -158,10 +158,9 @@ static int hda_link_dma_cleanup(struct snd_pcm_substream *substream,
 }
 
 static int hda_link_dma_hw_params(struct snd_pcm_substream *substream,
-				  struct snd_pcm_hw_params *params)
+				  struct snd_pcm_hw_params *params, struct snd_soc_dai *cpu_dai)
```

Comment:
this is also a bit controversial, in initial cleanups we made sure the DMA, DAI and IPC stuff were relatively isolated. I am not sold on the idea of dealing with DMA and DAI in the same functions.

Reply:
but you are dealing with them anyway. All I am saying use the pointer thats already available instead of letting it go and then looking it up again in this function

Reply:
The use of cpu_dai is only to store the stream tag/hext_stream information.

What really matters in this function is the codec dai which is use to determine what the link configuration will be.

I am not clear on the merit of adding a new parameter when we can follow the same approach for cpu_dai and codec_dai.

Is this to avoid looking up the cpu_dai multiple times?